### PR TITLE
EX-359: MSM encoders

### PR DESCRIPTION
**Pull request by nsirola**
Add constellation support to the MSM encoders. 

Add a GLO MSM5 unit test case, and augment the random data decoding cases so that whenever a random buffer is successfully decoded, test round-trip encoding and decoding on the resulting valid but random message. (note that the raw bit buffers will not be identical because some bytes will be left undefined, e.g. the fine range values corresponding to rough ranges marked invalid)

Correct functionality tested also in https://github.com/swift-nav/gnss-converters/pull/143

### CODE REVIEW

File: c/src/encode.c [hunk 1]
```diff
@@ -953,16 +1011,17 @@ static void encode_msm_fine_phaserangerates(const uint8_t num_cells,
                                             uint16_t *bit) {
   /* DF404 */
   for (uint16_t i = 0; i < num_cells; i++) {
-    if (flags[i].valid_dop) {
-      rtcm_setbits(buff, *bit, 15, (int16_t)(fine_range_rate_m_s[i] / 0.0001));
+    if (flags[i].valid_dop && fabs(fine_range_rate_m_s[i]) < 1.6383) {
```
Comment (pmiettinen) [inline c/src/encode.c]: Define for `1.6383`?

Comment (nsirola) [reply c/src/encode.c]: Used the power-of-two it's derived from.

File: c/src/encode.c [hunk 2]
```diff
@@ -15,6 +15,8 @@
 #include <math.h>
 #include <stdbool.h>
 #include <stdint.h>
+#include <stdio.h>
```
Comment (anth-cole) [inline c/src/encode.c]: is this left over from development?

File: c/src/encode.c [hunk 3]
```diff
@@ -976,11 +1035,14 @@ static uint16_t rtcm3_encode_msm_internal(const rtcm_msm_message *msg,
...
+    /* Unexpected message type. */
     return 0;
   }
 
-  if (RTCM_CONSTELLATION_GPS != to_constellation(msg->header.msg_num)) {
-    /* Unexpected message type. */
+  rtcm_constellation_t cons = to_constellation(header->msg_num);
+
+  if (RTCM_CONSTELLATION_INVALID == cons) {
+    fprintf(stderr, "Invalid constellation\n");
```
Comment (anth-cole) [inline c/src/encode.c]: This explains the stdio include - do we want this error handling? we could just ignore the invalid constellation

Comment (nsirola) [reply c/src/encode.c]: Removed `stdio` and printouts from this library and added an error printout on encoding error to `gnss-converters` instead.

**Pull request merged**